5.2.join.py

```diff
@@ -0,0 +1,18 @@
+def join(*params, sep = '-'):
+    if not params:
+        return None
+    L = []
+    for i in range(len(params)):
+        try:
+            iter(params[i])
+        except:
+            raise TypeError(params[i], 'is not iterable')
+        for j in range(len(params[i])):
+            L.append(params[i][j])
```

Comment:
1. Try to avoid i and j as well if there is a better name available.

2. Replace for loops -  
https://medium.com/python-pandemonium/never-write-for-loops-again-91a5a4c84baf
Replace for loops with more advanced python features